components/gitpod-db/src/user-to-team-migration-service.ts

```diff
@@ -89,7 +89,18 @@ export class UserToTeamMigrationService {
             newAttribution,
             oldAttribution,
         ]);
-        log.info(ctx, "Migrated cost center data.", { teamId: team.id, result });
+        if (result.affectedRows === 0) {
+            const now = new Date();
+            const nextMonth = new Date();
+            nextMonth.setMonth(now.getMonth() + 1);
+            await conn.query(
+                "INSERT INTO d_b_cost_center (id, creationTime, spendingLimit, billingStrategy, billingCycleStart, nextBillingTime) VALUES (?,?,?,?,?,?)",
+                [newAttribution, now.toISOString(), 500, "other", now.toISOString(), nextMonth.toISOString()],
+            );
+            log.info(ctx, "Created cost center data.", { teamId: team.id, result });
```

Comment:
FWIW, the intent was that `usage` is the only component which will actually perform DB operations on the `cost_center` and `usage` records. With this, we're now departing from that. There's significant risk from multiple components interacting with the same tables as maintaining data quality becomes difficult.

What's the reasoning for abandoning that principle?

Reply:
> What's the reasoning for abandoning that principle?

The reason I took this approach is simplicity. The alternative would have been introducing much more non-local code (i.e. new service and API in usage component) to allow this, which would also introduce another dependency and with that would make the test infrastructure more complex. I see that there is a chance that a change to the underlying d_b_cost_center could go unnoticed and result in bad data, I believe the likely hood of introducing bugs by adding the complexity mentioned above is higher.

Reply:
FWIW I was looking into doing this in GetOrCreateCostCenter, but I'm missing the user information there.

Reply:
What stops us from adding that to the API surface?

If we always take the easier path to do something, the cost of that doesn't go away, it just accrues interest and comes back to us later on.

Reply:
> If we always take the easier path to do something, the cost of that doesn't go away, it just accrues interest and comes back to us later on.

I believe it's not the easier path but the less costly (also long-term).